DigitalLearningSolutions.Data.Tests/Services/ActivityServiceTests.cs

```diff
@@ -344,6 +338,83 @@ private void GivenDataServicesReturnData(int centreId, int categoryId)
                 .Returns(courses);
         }
 
+        [Test]
+        public void GetActivityDataFileForCentre_returns_expected_excel_data()
+        {
+            // given
+            using var expectedWorkbook = new XLWorkbook(
+                TestContext.CurrentContext.TestDirectory + ActivityDataDownloadRelativeFilePath
+            );
+            GivenActivityDataServiceReturnsDataInExampleSheet();
+
+            var filterData = new ActivityFilterData(
+                DateTime.Parse("2020-9-1"),
+                DateTime.Parse("2021-9-1"),
+                null,
+                null,
+                null,
+                CourseFilterType.None,
+                ReportInterval.Months
+            );
+
+            // when
+            var resultBytes = activityService.GetActivityDataFileForCentre(101, filterData);
+
+            using var resultsStream = new MemoryStream(resultBytes);
+            using var resultWorkbook = new XLWorkbook(resultsStream);
+
+            // Then
+            using (new AssertionScope())
+            {
+                resultWorkbook.Worksheets.Count.Should().Be(expectedWorkbook.Worksheets.Count);
+                foreach (var resultWorksheet in resultWorkbook.Worksheets)
+                {
+                    var expectedWorksheet = expectedWorkbook.Worksheets.Worksheet(resultWorksheet.Name);
+                    var cells = resultWorksheet.CellsUsed();
+                    cells.Count().Should().Be(expectedWorksheet.CellsUsed().Count());
+
+                    foreach (var cell in cells)
+                    {
+                        var expectedCell = expectedWorksheet.Cell(cell.Address);
+                        cell.Value.Should().BeEquivalentTo(expectedCell.Value);
+                    }
+                }

```

Comment:
Could we pull this into a separate method or even a helper if it's useful in other Excel comparison tests.